odabot/cli.py

```diff
@@ -238,28 +274,91 @@ def update_workflow(last_commit,
+                                             engine=build_engine,
+                                             cleanup=cleanup,
+                                             namespace = deployment_namespace,
+                                             nb2wversion=os.environ.get('ODA_WF_NB2W_VERSION', nb2wver()))
+            except:
+                set_commit_state(project['id'], 
+                                 last_commit['id'], 
+                                 "build",
+                                 "failed",
+                                 description="ODA-bot unable to build the container. An e-mail with details has been sent.")
+                raise
+            else:
+                if container_info['image'].count('/') == 1:
+                    container_info['image'] 
+                    hub_url = f"https://hub.docker.com/r/{container_info['image'].split(':')[0]}"
+                else:
+                    hub_url = None # no universal way to construct clickable url 
+                set_commit_state(project['id'], 
+                                 last_commit['id'], 
+                                 "build",
+                                 "success",
+                                 description=(f"ODA-bot have successfully built the container in {(datetime.now() - bstart).seconds} seconds. "
+                                              f"Image pushed to registry as {container_info['image']}"),
+                                 target_url=hub_url)
+
+            #deploy
+            set_commit_state(project['id'], 
+                             last_commit['id'], 
+                             "deploy",
+                             "running",
+                             description="ODA-bot is deploying the workflow")
+            try:
+                workflow_name = project['name'].lower().replace(' ', '-').replace('_', '-') + '-workflow'
+                deployment_info = deploy_k8s(container_info,
+                                            workflow_name, 
+                                            namespace=deployment_namespace, 
+                                            check_live_through=dispatcher_deployment)    
+            except:
+                set_commit_state(project['id'], 
+                                 last_commit['id'], 
+                                 "deploy",
+                                 "failed",
+                                 description="ODA-bot unable to deploy the workflow. An e-mail with details has been sent.")
+                raise
+            else:
+                set_commit_state(project['id'], 
+                                 last_commit['id'], 
+                                 "deploy",
+                                 "success",
+                                 description=f"ODA-bot have successfully deployed {workflow_name} to {deployment_namespace} namespace")
+        
         except Exception as e:
```

Comment:
so this will include ContainerBuildException which reads ok? would be better to catch it explicitly, but ok